js/apps/admin-ui/src/components/users/UserSelect.tsx

```diff
@@ -53,46 +52,66 @@ export const UserSelect = ({
   const values: string[] | undefined = getValues(name!);
 
   const [open, toggleOpen, setOpen] = useToggle();
-  const [users, setUsers] = useState<(UserRepresentation | undefined)[]>([]);
+  const [selectedUsers, setSelectedUsers] = useState<UserRepresentation[]>([]);
+  const [searchedUsers, setSearchedUsers] = useState<UserRepresentation[]>([]);
   const [inputValue, setInputValue] = useState("");
   const [search, setSearch] = useState("");
   const textInputRef = useRef<HTMLInputElement>();
 
   const debounceFn = useCallback(debounce(setSearch, 1000), []);
 
   useFetch(
-    () => {
-      const params: UserQuery = {
-        max: 20,
-      };
+    async () => {
+      if (!values) {
+        return [];
+      }
+
+      const foundUsers = await Promise.all(
+        values.map((id) => adminClient.users.findOne({ id })),
+      );
+
+      return foundUsers.filter((user) => user !== undefined);
+    },
+    setSelectedUsers,
+    [values],
+  );
 
-      if (search) {
-        params.username = search;
+  useFetch(
+    async () => {
+      if (!search) {
+        return [];
       }
 
-      if (values?.length && !search) {
-        return Promise.all(
-          values.map((id: string) => adminClient.users.findOne({ id })),
-        );
+      const foundUsers = await adminClient.users.find({
+        username: search,
+        max: 20,
+      });
+
+      if (!values) {
+        return foundUsers;
       }
-      return adminClient.users.find(params);
+
+      return foundUsers.filter((user) => !values.includes(user.id!));
```

Comment:
It shouldn't remove selected users from the list, but rather show them as selected with a little blue tick:
![image](https://github.com/user-attachments/assets/f9bf9302-2014-42bb-b9b4-5aea2f574043)